src/CMakeLists.txt

```diff
@@ -0,0 +1,438 @@
+  COMMENT "Generating obj/build.h"
+  VERBATIM
+)
+add_library(bitcoin_clientversion OBJECT EXCLUDE_FROM_ALL
+  clientversion.cpp
+)
+target_compile_definitions(bitcoin_clientversion
+  PRIVATE
+    HAVE_BUILD_INFO
+)
+target_link_libraries(bitcoin_clientversion
+  PRIVATE
+    core_interface
+)
+add_dependencies(bitcoin_clientversion generate_build_info)
+
+add_subdirectory(crypto)
+add_subdirectory(univalue)
+add_subdirectory(util)
+if(WITH_MULTIPROCESS)
+  add_subdirectory(ipc)
+endif()
+
+#=============================
+# secp256k1 subtree
+#=============================
+message("")
+message("Configuring secp256k1 subtree...")
+set(SECP256K1_DISABLE_SHARED ON CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_ECDH OFF CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_RECOVERY ON CACHE BOOL "" FORCE)
+set(SECP256K1_BUILD_BENCHMARK OFF CACHE BOOL "" FORCE)
+set(SECP256K1_BUILD_TESTS ${BUILD_TESTS} CACHE BOOL "" FORCE)
+set(SECP256K1_BUILD_EXHAUSTIVE_TESTS ${BUILD_TESTS} CACHE BOOL "" FORCE)
+set(SECP256K1_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+include(GetTargetInterface)
+# -fsanitize and related flags apply to both C++ and C,
+# so we can pass them down to libsecp256k1 as CFLAGS.
+get_target_interface(core_sanitizer_cxx_flags "" sanitize_interface COMPILE_OPTIONS)
+set(SECP256K1_LATE_CFLAGS ${core_sanitizer_cxx_flags} CACHE STRING "" FORCE)
+unset(core_sanitizer_cxx_flags)
+# We want to build libsecp256k1 with the most tested RelWithDebInfo configuration.
+enable_language(C)
+foreach(config IN LISTS CMAKE_BUILD_TYPE CMAKE_CONFIGURATION_TYPES)
+  if(config STREQUAL "")
+    continue()
+  endif()
+  string(TOUPPER "${config}" config)
+  set(CMAKE_C_FLAGS_${config} "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
+endforeach()
+# If the CFLAGS environment variable is defined during building depends
+# and configuring this build system, its content might be duplicated.
+if(DEFINED ENV{CFLAGS})
+  deduplicate_flags(CMAKE_C_FLAGS)
+endif()
+set(CMAKE_EXPORT_COMPILE_COMMANDS OFF)
+add_subdirectory(secp256k1)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+string(APPEND CMAKE_C_COMPILE_OBJECT " ${APPEND_CPPFLAGS} ${APPEND_CFLAGS}")
+
+# Stable, backwards-compatible consensus functionality.
+add_library(bitcoin_consensus STATIC EXCLUDE_FROM_ALL
+  arith_uint256.cpp
+  consensus/merkle.cpp
+  consensus/tx_check.cpp
+  hash.cpp
+  primitives/block.cpp
+  primitives/transaction.cpp
+  pubkey.cpp
+  script/interpreter.cpp
+  script/script.cpp
+  script/script_error.cpp
+  uint256.cpp
+)
+target_link_libraries(bitcoin_consensus
+  PRIVATE
+    core_interface
+    bitcoin_crypto
+    secp256k1
+)
+
+if(WITH_ZMQ)
+  add_subdirectory(zmq)
+endif()
+
+# Home for common functionality shared by different executables and libraries.
+# Similar to `bitcoin_util` library, but higher-level.
+add_library(bitcoin_common STATIC EXCLUDE_FROM_ALL
+  addresstype.cpp
```

Comment:
Nit:
```patch
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	(revision 92f7fc21768b0d898655d68b5d76ff09ff71c682)
+++ b/src/CMakeLists.txt	(date 1723286445028)
@@ -106,8 +106,8 @@
   addresstype.cpp
   base58.cpp
   bech32.cpp
-  chainparamsbase.cpp
   chainparams.cpp
+  chainparamsbase.cpp
   coins.cpp
   common/args.cpp
   common/bloom.cpp
@@ -130,18 +130,18 @@
   key.cpp
   key_io.cpp
   merkleblock.cpp
+  net_permissions.cpp
   net_types.cpp
   netaddress.cpp
   netbase.cpp
-  net_permissions.cpp
   outputtype.cpp
   policy/feerate.cpp
   policy/policy.cpp
   protocol.cpp
   psbt.cpp
+  rpc/external_signer.cpp
   rpc/rawtransaction_util.cpp
   rpc/request.cpp
-  rpc/external_signer.cpp
   rpc/util.cpp
   scheduler.cpp
   script/descriptor.cpp
```

Reply:
See https://github.com/hebasto/bitcoin/pull/320.

Reply:
Resolved in https://github.com/bitcoin/bitcoin/pull/30454#issuecomment-2294786256.